Classes/Dataset.cs

```diff
@@ -65,8 +76,9 @@ async Task DownloadFilesAsync(string path)
 
         }
 
-        public void downloadAll(string path)
+        public void downloadAll(string path,int maxItems)
         {
+            counterHolder counter = new counterHolder() { Value = maxItems };
```

Comment:
Eventuell in dieser Klasse Member analog zu counter deklarieren

Reply:
Wie genau meinst du das?